test/remove.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var S = require('..');
+
+var eq = require('./internal/eq');
+
+
+test('remove', function() {
+
+  eq(typeof S.remove, 'function');
+  eq(S.remove.length, 2);
+  eq(S.remove.toString(), 'remove :: String -> StrMap a -> StrMap a');
+
+  function testBasic(key, strMap, strMapResult) {
+    eq(S.remove(key, strMap), strMapResult);
+  }
+
+  function testDoubleRemove(key, strMap) {
+    var remove = S.remove(key);
+    eq(remove(remove(strMap)), remove(strMap));
+  }
```

Comment:
Let's replace this with a property-based test:

```javascript
jsc.assert(jsc.forall(jsc.string, jsc.dict(jsc.number), function(key, map) {
  var remove = S.remove(key);
  return S.equals(remove(remove(map)), remove(map));
}), {tests: 1000});
```

You should be able to do something similar for `testDoubleInsert`. :)

Reply:
Was unaware of jsverify! Very nice. I've made the changes as well as the above `Point` inherited properties test. Note that it does take well over triple the time to run the tests now on my MacBook due to insert.js taking 4.5secs and remove.js taking 3.5secs with these jsverify tests.